pkg/secrets/strategy.go

```diff
@@ -130,18 +116,84 @@ func (s Source) MarshalYAML() (interface{}, error) {
 	return s.MarshalRaw(), nil
 }
 
-type StrategyList []SourceMap
+// SourceMapList is a list of mappings that can be access via index or the item name.
+type SourceMapList []SourceMap
 
-func (l StrategyList) Less(i, j int) bool {
+func (l SourceMapList) Less(i, j int) bool {
 	return l[i].Name < l[j].Name
 }
 
-func (l StrategyList) Swap(i, j int) {
+func (l SourceMapList) Swap(i, j int) {
 	tmp := l[i]
 	l[i] = l[j]
 	l[j] = tmp
 }
 
-func (l StrategyList) Len() int {
+func (l SourceMapList) Len() int {
 	return len(l)
 }
+
+// HasName determines if the specified name is defined in the set.
+func (l SourceMapList) HasName(name string) bool {
+	_, ok := l.GetByName(name)
+	return ok
+}
+
+// GetByName returns the resolution strategy for the specified name and a bool indicating if it was found.
+func (l SourceMapList) GetByName(name string) (SourceMap, bool) {
+	for _, item := range l {
+		if item.Name == name {
+			return item, true
+		}
+	}
+
+	return SourceMap{}, false
+}
+
+// GetResolvedValue returns the resolved value of the specified name and a bool indicating if it was found.
+// You must resolve the value before calling, it does not do resolution for you.
+func (l SourceMapList) GetResolvedValue(name string) (interface{}, bool) {
+	item, ok := l.GetByName(name)
+	if ok {
+		return item.ResolvedValue, true
+	}
+
+	return nil, false
+}
+
+// GetResolvedValues converts the items to a map of key/value pairs, with the values represented as CNAB-compatible strings
+func (l SourceMapList) GetResolvedValues() map[string]string {
```

Comment:
I'm going to avoid `GetValue()` because it's important to call out that it's a resolved value, i.e. Porter called out to a secret store or read an environment variable using the strategy defined for that item, and this is the final value.

But I agree it would be easy to mistype. I'll rename it to `ToResolvedValues` so that it emphasizes that it's returning a new struct (not just looking up data like the other Get functions).